x-pack/plugins/task_manager/server/saved_objects/migrations.test.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import uuid from 'uuid';
+import { getMigrations } from './migrations';
+import { SavedObjectUnsanitizedDoc } from 'kibana/server';
+import { migrationMocks } from 'src/core/server/mocks';
+import { TaskInstanceWithDeprecatedFields } from '../task';
+
+const migrationContext = migrationMocks.createContext();
+
+describe('successful migrations', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+  describe('7.4.0', () => {
+    test('extend task instance with updated_at', () => {
+      const migration740 = getMigrations()['7.4.0'];
+      const taskInstance = getMockData({});
+      expect(migration740(taskInstance, migrationContext).attributes.updated_at).not.toBeNull();
+    });
+  });
+
+  describe('7.6.0', () => {
+    test('rename property Internal to Schedule', () => {
+      const migration760 = getMigrations()['7.6.0'];
+      const taskInstance = getMockData({});
+      expect(migration760(taskInstance, migrationContext)).toEqual({
+        ...taskInstance,
+        attributes: {
+          ...taskInstance.attributes,
+          schedule: taskInstance.attributes.schedule,
+        },
+      });
+    });
+  });
+
+  describe('8.0.0', () => {
+    test('transforms actionsTasksLegacyIdToSavedObjectIds', () => {
+      const migration800 = getMigrations()['8.0.0'];
+      const taskInstance = getMockData({
+        taskType: 'actions:123456',
+        params: JSON.stringify({ spaceId: 'user1', actionTaskParamsId: '123456' }),
+      });
+
+      expect(migration800(taskInstance, migrationContext)).toEqual({
+        ...taskInstance,
+        attributes: {
+          ...taskInstance.attributes,
+          params: '{"spaceId":"user1","actionTaskParamsId":"800f81f8-980e-58ca-b710-d1b0644adea2"}',
+        },
+      });
+    });
+
+    test('it is only applicable for saved objects that live in a custom space', () => {
```

Comment:
When spaces are disabled, the `spaceId` is not included in the params. Can we add a test for this scenario for actions and alerting just to cover all our bases?